tests/components/group/test_init.py

```diff
@@ -367,72 +320,150 @@ def test_group_updated_after_device_tracker_zone_change(self):
-            "homeassistant.config.load_yaml_config_file",
-            return_value={
-                "group": {"hello": {"entities": "light.Bowl", "icon": "mdi:work"}}
-            },
-        ):
-            common.reload(self.hass)
-            self.hass.block_till_done()
-
-        assert sorted(self.hass.states.entity_ids()) == [
-            "group.all_tests",
-            "group.hello",
-        ]
-        assert self.hass.bus.listeners["state_changed"] == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["light.bowl"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.one"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.two"]) == 1
-
-    def test_modify_group(self):
-        """Test modifying a group."""
-        group_conf = OrderedDict()
-        group_conf["modify_group"] = {"name": "friendly_name", "icon": "mdi:work"}
-
-        assert setup_component(self.hass, "group", {"group": group_conf})
-
-        # The old way would create a new group modify_group1 because
-        # internally it didn't know anything about those created in the config
-        common.set_group(self.hass, "modify_group", icon="mdi:play")
-        self.hass.block_till_done()
+    assert group.is_on(hass, "group.none") is False
+    assert await async_setup_component(hass, "light", {})
+    assert await async_setup_component(hass, "group", {})
+    await hass.async_block_till_done()
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling"], False
+    )
+    await hass.async_block_till_done()
+
+    assert group.is_on(hass, test_group.entity_id) is True
+    hass.states.async_set("light.Bowl", STATE_OFF)
+    await hass.async_block_till_done()
+    assert group.is_on(hass, test_group.entity_id) is False
+
+    # Try on non existing state
+    assert not group.is_on(hass, "non.existing")
+
+
+async def test_reloading_groups(hass):
+    """Test reloading the group config."""
+    assert await async_setup_component(
+        hass,
+        "group",
+        {
+            "group": {
+                "second_group": {"entities": "light.Bowl", "icon": "mdi:work"},
+                "test_group": "hello.world,sensor.happy",
+                "empty_group": {"name": "Empty Group", "entities": None},
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    await group.Group.async_create_group(
+        hass, "all tests", ["test.one", "test.two"], user_defined=False
+    )
+
+    await hass.async_block_till_done()
+
+    assert sorted(hass.states.async_entity_ids()) == [
+        "group.all_tests",
+        "group.empty_group",
+        "group.second_group",
+        "group.test_group",
+    ]
+    assert hass.bus.async_listeners()["state_changed"] == 1
+    assert len(hass.data[TRACK_STATE_CHANGE_CALLBACKS]["hello.world"]) == 1
+    assert len(hass.data[TRACK_STATE_CHANGE_CALLBACKS]["light.bowl"]) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6259301)